includes/Modules/Analytics/Advanced_Tracking/Measurement_Code_Injector.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Class Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Code_Injector
+ *
+ * @package   Google\Site_Kit\Modules\Analytics
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Analytics\Advanced_Tracking;
+
+/**
+ * Injects Javascript based on the current active plugins
+ *
+ * Class Injector
+ *
+ * @since n.e.x.t.
+ * @access private
+ * @ignore
+ */
+final class Measurement_Code_Injector {
+
+	/**
+	 * Holds a list of event configurations to be injected
+	 *
+	 * @since n.e.x.t.
+	 * @var array
+	 */
+	private $event_configurations;
+
+	/**
+	 * The javascript code that is injected for event tracking.
+	 *
+	 * @since n.e.x.t.
+	 * @var string
+	 */
+	private static $inject_script = <<<INJECT_SCRIPT
+let config;
+for ( config of eventConfigurations ) {
+	const thisConfig = config;
+	document.addEventListener( config.on, function( e ) {
+		if ( e.target.matches( thisConfig.selector ) ) {
+			alert( 'Got an event called: '.concat( thisConfig.action ) );
+			//record event with gtag here
+		} else if ( e.target.matches( thisConfig.selector.concat( ' *' ) ) ) {
+			alert( 'Got an event called: '.concat( thisConfig.action ) );
+			//record event with gtag here
```

Comment:
I think we should now start adding the actual `gtag` calls here. `gtag` will be present in the page since the Analytics script from Site Kit is present at this point.